web/api/app/Controller/AppController.php

```diff
@@ -63,9 +63,47 @@ public function beforeFilter() {
         	$zmOptAuth = $config['Config']['Value'];
         	if (!$this->Session->Read('user.Username') && ($zmOptAuth=='1'))
         	{       
-        		 throw new NotFoundException(__('Not Authenticated'));
+        		 throw new UnauthorizedException(__('Not Authenticated'));
         	 	return; 
         	}     
+		// PP #1155
+		else
+		{
+			$this->loadModel('User');
+			$loggedinUser = $this->Session->Read('user.Username');
+			$isEnabled = $this->Session->Read('user.Enabled');
+			// PP - this will likely never happen as if its
+			// not enabled, login will fail and Not Auth will be returned
+			// however, keeping this here for now
+			if ($isEnabled != "1")
+			{
+				throw new UnauthorizedException(__('User is not enabled'));
+				return;
+			}
+
+			if ($zmOptAuth=='1')
+			{
+				$options = array ('conditions' => array ('User.Username' => $loggedinUser));
+				$userMonitors = $this->User->find('first', $options);
+				$this->Session->Write('allowedMonitors',$userMonitors['User']['MonitorIds']);
```

Comment:
@kylejohnson , based on my reading:

``Configure::write`` is for application wide variables while ``Session->Write`` is for settings that get set only for that specific session. The latter is what we need for these variables as they will change depending on who has opened the session